packages/gatsby-source-filesystem/src/gatsby-node.js

```diff
@@ -102,6 +103,12 @@ const createFSMachine = (
           }
           createAndProcessNode(path).catch(err => reporter.error(err))
         },
+        updateAndProcessNode(_, { pathType, path }, { state }) {
```

Comment:
Second param to xstate actions should also have `type`. Given that only thing that changes between `createAndProcessNode` and `updateAndProcessNode` is log (added / changed) - maybe we could have single action that changes that part based on action type to not repeat pretty much same action? Similarly with `queueNodeCreation` and `queueNodeUpdating` we could conditionally figure out which `op` to use

Reply:
Using the proper xstate way by creating another side-effect.